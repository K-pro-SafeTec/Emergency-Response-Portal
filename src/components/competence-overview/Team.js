import React from 'react';
import { teamById } from '../../dummy-data/team';
import { personById } from '../../dummy-data/person';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import StatusIcon from '../shared/StatusIcon';
import AppPage from '../shared/AppPage';

export default ({ match }) => {
  const team = teamById[match.params.teamId];
  return team ? (
    <AppPage title={team.name}>
      <List>
        {team.members.map(memberId => {
          const person = personById[memberId];
          return person ? (
            <ListItem key={memberId} button>
              <ListItemIcon style={{ fontSize: '40px' }}>
                <StatusIcon status={team.status} />
              </ListItemIcon>
              <ListItemText primary={person.name} />
            </ListItem>
          ) : (
            <ListItem key={memberId}>
              <ListItemText>Ukjent person</ListItemText>
            </ListItem>
          );
        })}
      </List>
    </AppPage>
  ) : (
    <p>Fant ikke noe lag med denne IDen.</p>
  );
};
