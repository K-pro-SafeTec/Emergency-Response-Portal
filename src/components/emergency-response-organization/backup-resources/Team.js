import React, {Component} from 'react';
import Person from "./Person";

export default class Team extends Component {
  render() {
    return(
      <div style={{marginBottom: '32px'}}>
        <h2>{this.props.name}</h2>
        {this.props.backups.map(({ id, name, status }) => (
          <Person key={id} name={name} status={status}/>
        ))}
      </div>
    )
  }
}
