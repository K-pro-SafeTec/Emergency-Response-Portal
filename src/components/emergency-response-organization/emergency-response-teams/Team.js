import React from 'react';
import Paper from '@material-ui/core/Paper';
import Person from "./Person";
import { personById } from "../../../dummy-data/person"

const styles = {
  margin: '1em',
  padding: '1em',
  width: '17em'
};

const Team = ({team}) => {
  const teamMembers = team.members.map(memberId => personById[memberId]);
  return(
    <Paper style={styles}>
      <h2>{team.name}</h2>
      {teamMembers.map(member => (
        <Person key={member.id} name={member.name}/>
      ))}
    </Paper>
  )
};

export default Team;
