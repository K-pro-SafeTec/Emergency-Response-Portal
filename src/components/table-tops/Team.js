import React from 'react';
import Typography from '@material-ui/core/Typography';
import AppPage from '../shared/AppPage';
import { tableTopInstanceList } from '../../dummy-data/tableTopInstance';
import { teamById } from '../../dummy-data/team';
import { tableTopScenarioById } from '../../dummy-data/tableTopScenario';
import TableTopTable from "./TableTopTable";


const Team = ({ match }) => {
  const team = teamById[match.params.teamId];
  if (team) {
    const instances = tableTopInstanceList.filter(instance => tableTopScenarioById[instance.scenario].team === team.id);
    return (
      <AppPage title="Table top" back="..">
        <TableTopTable
          instances={instances}
        />
      </AppPage>
    );
  }
  return (
    <AppPage title="Table top" back="..">
      <Typography>Fant ikke noe lag med denne IDen</Typography>
    </AppPage>
  )
};

export default Team;
