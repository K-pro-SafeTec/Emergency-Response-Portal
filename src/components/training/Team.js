import React from 'react';
import Typography from '@material-ui/core/Typography';
import AppPage from '../shared/AppPage';
import TrainingTable from './TrainingTable';
import { trainingInstanceList } from '../../dummy-data/trainingInstance';
import { teamById } from '../../dummy-data/team';
import { trainingScenarioById } from '../../dummy-data/trainingScenario';


const Team = ({ match }) => {
  const team = teamById[match.params.teamId];
  if (team) {
    const instances = trainingInstanceList.filter(instance => trainingScenarioById[instance.scenario].teams.indexOf(team.id) !== -1);
    return (
      <AppPage title="Trening" back="..">
        <TrainingTable
          instances={instances}
        />
      </AppPage>
    );
  }
  return (
    <AppPage title="Trening" back="..">
      <Typography>Fant ikke noe lag med denne IDen</Typography>
    </AppPage>
  )
};

export default Team;
