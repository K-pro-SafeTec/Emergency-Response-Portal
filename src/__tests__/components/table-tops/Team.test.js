import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom'
import Team from "../../../components/table-tops/Team";

let testRenderer;
let testInstance;

beforeAll(() => {
  testRenderer = TestRenderer.create(
    <Router>
      <Team
        match={{
          params: {
            teamId: 1,
          }
        }}
      />
    </Router>
  );
  testInstance = testRenderer.root;
});

test('renders correctly', () => {
  const tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});
