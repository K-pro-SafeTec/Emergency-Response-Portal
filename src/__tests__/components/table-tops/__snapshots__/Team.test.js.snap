// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-14 MuiPaper-elevation4-20 MuiAppBar-root-5 MuiAppBar-positionFixed-6 MuiAppBar-colorDefault-11 mui-fixed"
  >
    <div
      className="MuiToolbar-root-41 MuiToolbar-regular-43 MuiToolbar-gutters-42"
    >
      <a
        className="MuiButtonBase-root-51 MuiIconButton-root-45 AppPage-homeButton-2"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-115"
        />
      </a>
      <button
        className="MuiButtonBase-root-51 MuiButtonBase-disabled-52 MuiIconButton-root-45 MuiIconButton-disabled-49"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
      </button>
      <button
        className="MuiButtonBase-root-51 MuiButtonBase-disabled-52 MuiIconButton-root-45 MuiIconButton-disabled-49"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </span>
      </button>
      <h2
        className="MuiTypography-root-63 MuiTypography-title-69 MuiTypography-colorInherit-92 AppPage-grow-1"
      >
        Table tops
      </h2>
      <a
        className="AppPage-safetecLink-3"
        href="https://www.safetec.no/"
      >
        <img
          alt="Levert av Safetec"
          height="44px"
          src="safetec.svg"
          width="153px"
        />
      </a>
    </div>
  </header>,
  <div
    className="AppPage-content-4"
  >
    <div
      className="MuiTable-root-98"
    >
      <div
        className="MuiTableHead-root-99"
      >
        <div
          className="MuiTableRow-root-100 MuiTableRow-head-103"
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-head-106"
            scope="col"
          >
            Dato
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-head-106"
            scope="col"
          >
            DFU
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-head-106"
            scope="col"
          >
            Tittel
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-head-106"
            scope="col"
          >
            Scenario
          </div>
        </div>
      </div>
      <div
        className="MuiTableBody-root-113"
      >
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/4/2018-09-24/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2018-09-24
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            4
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Fare for kollisjon
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Drivende fartøy på kollisjonskurs
          </div>
        </a>
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/4/2018-10-08/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2018-10-08
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            4
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Fare for kollisjon
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Drivende fartøy på kollisjonskurs
          </div>
        </a>
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/4/2018-10-22/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2018-10-22
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            4
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Fare for kollisjon
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Drivende fartøy på kollisjonskurs
          </div>
        </a>
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/3/2018-11-05/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2018-11-05
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            3
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Akutt utslipp
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Oljeutslipp til sjø fra innretninig
          </div>
        </a>
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/3/2018-11-19/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2018-11-19
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            3
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Akutt utslipp
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Oljeutslipp til sjø fra innretninig
          </div>
        </a>
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/3/2018-12-03/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2018-12-03
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            3
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Akutt utslipp
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Oljeutslipp til sjø fra innretninig
          </div>
        </a>
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/12/2018-12-17/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2018-12-17
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            12
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Kriminell handling
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Bombetrussel
          </div>
        </a>
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/12/2018-12-31/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2018-12-31
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            12
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Kriminell handling
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Bombetrussel
          </div>
        </a>
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/12/2019-01-14/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2019-01-14
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            12
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Kriminell handling
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Bombetrussel
          </div>
        </a>
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/14/2019-01-28/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2019-01-28
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            14
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Tap av stabilitet
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Vanninntrengning
          </div>
        </a>
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/14/2019-02-11/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2019-02-11
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            14
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Tap av stabilitet
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Vanninntrengning
          </div>
        </a>
        <a
          className="MuiTableRow-root-100 MuiTableRow-hover-102 LinkTableRow-noUnderline-114"
          href="/scenarios/14/2019-02-25/"
          onClick={[Function]}
        >
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            2019-02-25
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            14
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Tap av stabilitet
          </div>
          <div
            aria-sort={null}
            className="MuiTableCell-root-105 MuiTableCell-body-107"
          >
            Vanninntrengning
          </div>
        </a>
      </div>
    </div>
  </div>,
]
`;
